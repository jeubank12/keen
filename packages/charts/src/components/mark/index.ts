import Mark, { markMotion } from './mark.component';

export { Mark, markMotion };
