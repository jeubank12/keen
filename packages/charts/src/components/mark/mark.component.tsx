import React, { memo } from 'react';

import { colors } from '@keen.io/colors';

type Props = {
  x: number;
  y: number;
  color: string;
  outerRadius?: number;
  innerRadius?: number;
  strokeWidth?: number;
};

export const markMotion = {
  initial: { opacity: 0.3, scale: 0.2 },
  animate: { opacity: 1, scale: 1 },
  transition: { duration: 0.3 },
  exit: { opacity: 0, scale: 0.3 },
};

const Mark = ({
  color,
  x,
  y,
  outerRadius = 10,
  innerRadius = 4,
  strokeWidth = 4,
}: Props) => (
  <g pointerEvents="none">
    <circle
      cx={x}
      cy={y}
      r={outerRadius}
      stroke={colors.white['500']}
      strokeWidth={strokeWidth}
      fill={color}
    />
    <circle cx={x} cy={y} r={innerRadius} fill={colors.white['500']} />
  </g>
);

export default memo(Mark);
