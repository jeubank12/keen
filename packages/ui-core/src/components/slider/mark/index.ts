import Mark from './mark.component';

export default Mark;
