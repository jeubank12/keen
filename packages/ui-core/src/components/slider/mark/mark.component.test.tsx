import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Mark from './mark.component';
import { Circle } from './mark.styles';

describe('@keen.io/ui-core - <Mark />', () => {
  const props = {
    size: 10,
    backgroundColor: 'black',
    borderColor: 'gray',
  };

  it('should call "onMouseEnter" handler', () => {
    const mockFn = jest.fn();
    const wrapper = mount(<Mark {...props} onMouseEnter={mockFn} />);

    wrapper.simulate('mouseEnter');

    expect(mockFn).toHaveBeenCalled();
  });

  it('should call "onMouseLeave" handler', () => {
    const mockFn = jest.fn();
    const wrapper = mount(<Mark {...props} onMouseLeave={mockFn} />);

    wrapper.simulate('mouseLeave');

    expect(mockFn).toHaveBeenCalled();
  });

  it('should set element dimension based on "size" property', () => {
    const wrapper = mount(<Mark {...props} />);

    expect(wrapper.find(Circle)).toMatchSnapshot();
  });
});
