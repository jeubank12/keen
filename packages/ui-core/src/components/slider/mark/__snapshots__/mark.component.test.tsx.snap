// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@keen.io/ui-core - <Mark /> should set element dimension based on "size" property 1`] = `
.c0 {
  box-sizing: border-box;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background: black;
  border: solid 2px gray;
}

<styled.div
  backgroundColor="black"
  borderColor="gray"
  size={10}
>
  <StyledComponent
    backgroundColor="black"
    borderColor="gray"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  box-sizing: border-box;
  width: ",
            [Function],
            "px;
  height: ",
            [Function],
            "px;
  border-radius: 50%;
  background: ",
            [Function],
            ";
  border: solid 2px ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    size={10}
  >
    <div
      className="c0"
      size={10}
    />
  </StyledComponent>
</styled.div>
`;
